### ISSUE TITLE
Sorting movies by recently added does not actually sort by most recently added

### ISSUE THREAD

**Issue by fuzxi** [jellyfin/jellyfin#6178]
**Describe the bug**

Sorting by date added, as well as the "Latest Movies" section on the home page, does not sort by date added. If I run `ls -lt /data/media/movies` to list my movies folder by date, the three most recent movies are Splice, Super, and Antichrist. If I sort my movie library by date, the first four movies are Cool Hand Luke, Splice, Memento, and Super. Antichrist is somewhere in the middle of my collection. Cool Hand Luke and Memento have creation dates in 2020. Cool Hand Luke is about the 30th most recent movie, Memento is about the 20th most recent.

Additionally, the Latest Movies section on the home page does not behave the same way. It shows none of these five movies. It instead shows an apparently random collection of somewhat recent movies which does not change when I add new movies.

This problem does _not_ occur with my Series library which is hosted on the same remote mount; both the library view and Latest Series section are sorted as expected and identically. 

Disabling external plugins and restarting did not have an effect on this issue.

**System (please complete the following information):**
 - OS: DietPi on RasPi4
 - Virtualization: Docker (jellyfin/jellyfin:latest)
 - Clients: Browser and Android client
 - Browser: Firefox 89.0
 - Jellyfin Version: 10.7.5
 - Playback: N/A
 - Installed Plugins: AudioDB, Bookshelf, MusicBrainz, OMDb, TMDb Box Search, TheTVDB
 - Reverse Proxy: None
 - Base URL: None
 - Networking: Docker bridge network
 - Storage: Cloud (GSuite/Google Drive) via read-only rclone mount

**To Reproduce**

1. Open Movies library
2. Click sort
3. Select "Date Added"
4. Observe that movies are not sorted by date added

**Expected behavior**

Library should be sorted by most recently added.

**Server Logs**

[log.txt](https://github.com/jellyfin/jellyfin/files/6643381/log.txt)

**FFmpeg Logs**
N/A

**Browser Console Logs**
N/A
**Screenshots**

N/A
**Additional context**

N/A

**Comment by michael9dk** [comment]
Could the issue be the GetDate method here. It returns DateTime.MinValue for non-folders, that are used for comparison?

https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Sorting/DateLastMediaAddedComparer.cs

**Comment by michael9dk** [comment]
@fuzxi 
Do you store the movies in subfolders inside /data/media/movies ?

**Comment by fuzxi** [comment]
> @fuzxi
> Do you store the movies in subfolders inside /data/media/movies ?

Yes. For example, Memento's path would be `/data/media/movies/Memento (2000)/Memento (2000) Remux-1080p.mkv`

**Comment by martadinata666** [comment]
So when i think my sorted videos are weird, it seems jellyfin not recursively check video in folder.

`Folder A created 15 Feb and Video today, it still use Folder A created date as sort date, not the file added in.`

**Comment by PAHelper2** [comment]
FYI I've noticed this behavior forever also.  I'm running 10.7.something on a Synology NAS, and I have no idea how "Date added" sort is actually figured out.

**Comment by hcgonzalezpr** [comment]
I'm able to reproduce this behavior on Version: 10.8.11. What additional information is required?

**Comment by psaintmalo** [comment]
Can reproduce in 10.8.12
Movie libraries when sorted by "Date Added" appear to be sorted by the **release date** of the media.
For other libraries, sorting by "Date Added" will sort them by the date that the media was added to the library.

**Comment by martadinata666** [comment]
Please stop closing issue that unsolved or stale, due no response from developer.

**Comment by oddstr13** [comment]
@martadinata666 If you comment within 3 weeks after the first stalebot message, the four month stale timer get reset.

**Comment by martadinata666** [comment]
up

**Comment by patcfly** [comment]
I am experiencing this bug - always have. I store media in their own directories under the movies dir. Additionally, I use mergerfs to present multiple volumes as a single volume to my docker services. Not that I think that matters. 

What I experience is that after every pull and restart via compose up -d, the date added sort changes. Some of my most recently added files look like they were added first and some of my first added media appear like I just added them. 

This is particularly annoying when using a service like jellyseer that lets users request media, it completes, and then it doesnt show up when browsing jellyfin as recently added content.

**Comment by barndawe** [comment]
I'm having the same issue. The movie files are being put into named folders for each single movie and not appearing in the order they were added. This exact same setup works just fine on Plex

**Comment by JkBoyo** [comment]
So my instance seems to be working fine based on file creation date. Can you verify that your setting ![Image](https://github.com/user-attachments/assets/a14a5721-9d4a-4e75-9892-bf44c36dec2f)
is set to use file creation date or the date added to library?
It could be we just need to better let people know that this setting exists.

**Comment by barndawe** [comment]
I've just tried changing it, including restarting the server afterwards. Both settings give me exactly the same list, and still not in order of when they were added. Currently running 10.9.11 via Docker on a Debian VM host

**Comment by patcfly** [comment]
My setting matches yours:
![image](https://github.com/user-attachments/assets/5debd130-1a0b-421c-9a25-48676274b73c)

**Comment by JkBoyo** [comment]
interesting. I will have to do some playing around and see where the location of the data it's fetching is held. Bc it does look like in mine that they match this.  Did you guys check using ls -lt on the dirs where you have it stored to compare those to the order presented by the lib like OP? And maybe instead of a server restart try a library scan update?

**Comment by barndawe** [comment]
`ls -lt` on my movie dir gives me the order I'd expect (and the order I get in P{lex), not the order that Jellyfin is giving me. I tried library scan update as well,forgot to say. It made no difference sadly.

**Comment by JkBoyo** [comment]
Do the metadata dates on the individual pieces of content match the ordering? For example 
![image](https://github.com/user-attachments/assets/1fa344cd-1637-4592-b785-d7536e3c5d3d)

This date on mine doesn't match the actual file creation date.

**Comment by JkBoyo** [comment]
but the order is still the same.

**Comment by barndawe** [comment]
The 'date added' does match the ordering, and also matches the file created date ordering, but not the directories they are within. i.e. I have one file that claims it was created in 2007, but it was added a few months ago and the containing folder's creation date reflects that. 
![image](https://github.com/user-attachments/assets/27b96b31-248e-44ad-a9d5-2fc8eb45bcd7)
As you can see above I've got library added date for the date and forced both a library rescan and a metadata regen, no change.

**Comment by JkBoyo** [comment]
So you think it is ordering things based on when the directory was created?

**Comment by barndawe** [comment]
Ordering seems to be based on the 'file created date' even when 'use date scanned into the library' is selected for me, and the file created dates are wrong. It looks like Plex is using the date the folder containing the file was created (my library is like 'movies/movie name/movie name.mp4', for example) but Jellyfin is using file date even when it's told not to

**Comment by JkBoyo** [comment]
So that setting seems to be how it *adds* the metadata field for new content not how it pulls it in during a library scan. Can you test to see if when set to `scan in date` it uses that for new content?

**Comment by martadinata666** [comment]
Nothing settings will matter, it just jellyfin love using "folder created" add instead of file added in to folder, this behaviour more seen when we collect many random videos.
```
/videos/memevid/yyy.mp4 (let say video `yyy.mp4` added april 2024, folder created dec 2023)
/videos/www.mp4 (added feb 2024)
```
In theory by "recently added" ```yyy.mp4``` should be sorted first. But no, jellyfin will use ```www.mp4``` as the first recently added. Doesn't matter the setting it just work that way.

**Comment by JkBoyo** [comment]
So @maradinata666 your comment is now contradicting @barndawe 's comment. I understand we have an issue here and I am trying to help prove it is something we can put our time to. To do that though I have to be able to fully understand and reproduce the issue.  Are you positive it will always work the same way and ignore the aforementioned setting? Have you tested it?

**Comment by martadinata666** [comment]
> So @maradinata666 your comment is now contradicting @barndawe 's comment. I understand we have an issue here and I am trying to help prove it is something we can put our time to. To do that though I have to be able to fully understand and reproduce the issue. Are you positive it will always work the same way and ignore the aforementioned setting? Have you tested it?

Yes, I'm deleting,rescan, or change many sort settings, etc. It just don't get sorting "recently added" right, plex and emby done right. IMO Jellyfin use the `folder creation date` instead of the file added in to the folder, that the file may got different "creation date", "added date" or "bla bla date" that making jellyfin sort correctly. Maybe the most correct term will be "recently added" jellyfin did not use a flat directory structure, which most user expects. If the folder was created in 2023, we need to scroll down the folder, and click the folder, the newly added video will show the most top left (right behavior). Why? When I click my library just show my media that I added last time, I don't want to scroll to find my folder to access my newly added media. Let say jellyfin don't want use "flat directory" then the folder that created 2023 should shown most top left as a media added in it recently.

**Comment by stefandxm** [comment]
So that setting is a bit silly for default. Newly added should be new added not newly created on disk imho. 

After changing the setting and making a re-scan, it is not automatically fixed. Can that be arranged? It wont even re-fix the manually touched files.

**Comment by cmck359** [comment]
I encountered this error while running version 10.10.5.  

Upon inspecting the `library.db`, it appears that when a new child item is added to an entity—for example, when a new episode is scanned during a library refresh—the parent entity (such as the TV series) does not have its `DateLastMediaAdded` field updated.  From what I can tell the sort order matches the `DateLastMediaAdded` 1:1.

The `DateLastMediaAdded` value is updated when performing a "Scan for missing metadata" refresh. However, it would be ideal if this also occurred during a "Scan for new and updated files" refresh.

Also, `DateLastMediaAdded` does not appear to be user-changeable, as it does not appear in the jellyfin web metadata editor.

**Comment by death69inc** [comment]
So what your saying is it needs to be fixed be the Dev team.

**Comment by felix920506** [comment]
This is a user configurable setting.

Users with admin privileges on the Jellyfin server can change it at: Dashboard > Libraries > Display

**Comment by patcfly** [comment]
I see it is admin configurable @felix920506  the problem was that it was not working as the UI lead us to believe. It is actually working as expected in 10.10.7 though. I don't know when it started working as expected but it was recent.
